source/extensions/filters/http/common/jwks_fetcher.cc

```diff
@@ -47,7 +47,7 @@ class JwksFetcherImpl : public JwksFetcher,
     // Check if cluster is configured, fail the request if not.
     const auto thread_local_cluster = cm_.getThreadLocalCluster(remote_jwks_.http_uri().cluster());
     if (thread_local_cluster == nullptr) {
-      ENVOY_LOG(error, "{}: fetch pubkey [uri = {}] failed: [cluster = {}] is not configured",
+      ENVOY_LOG(debug, "{}: fetch pubkey [uri = {}] failed: [cluster = {}] is not configured",
```

Comment:
Not,  I feel that this change will make it very difficult to find the root cause if "cluster" name is wrong.  Usually, envoy log is not at debug level.  

I feel that having error level making it easier to find the root cause.

Reply:
For root causing, we should attach the error detail on the request itself. The issue is that this makes logs explode when the error happens, and is inconsistent with other RDS/LDS cases with the missing cluster.

Reply:
Normally, Envoy doesn't log per request (that's reserved for debug/tracing levels). In this case, it logs an error for every request, and that explodes with high traffic.

Inconsistency is allowed in xDS, routes can reference missing clusters, etc. Envoy doesn't log an error then, but adds a detail per request, see `cluster_not_found` response code detail.

Reply:
Typically such error is attaches as the reason string when rejecting the request (assuming that in this case request is rejected). In this way it will be logged in the access log.